### PR TITLE
Rewrite using es6

### PR DESCRIPTION
**Pull request by gartz**
Change the syntax from ES3 to ES6 since we're using Babel, this syntax is shorter and easier to work with.

### CODE REVIEW

File: src/push-state-tree.js [hunk 1]
```diff
@@ -724,22 +746,20 @@ if (!PushStateTree.prototype.pushState) {
...
 
     return this;
   };
-}
 
-if (!PushStateTree.prototype.replaceState) {
-  PushStateTree.prototype.replaceState = function(state, title, uri) {
-    var t = document.title || '';
+  PushStateTree.prototype.replaceState = function (state, title, uri) {
+    title = title || document.title || '';
```
Comment (romuloab) [inline src/push-state-tree.js]: I don't think document.title can be anything but a string

File: src/push-state-tree.js [hunk 2]
```diff
@@ -682,30 +706,28 @@ function preProcessUriBeforeExecuteNativeHistoryMethods(method) {
...
     // Chainnable
     return this;
   };
 }
 
 // Wrap history methods
-for (var method in root.history) {
-  if (typeof root.history[method] === 'function') {
+for (var method in history) {
+  if (typeof history[method] === 'function') {
```
Comment (romuloab) [inline src/push-state-tree.js:717]: maybe check for `history.hasOwnProperty(method)` too?

File: src/push-state-tree.js [hunk 3]
```diff
@@ -317,331 +334,338 @@ var eventsQueue = [];
...
+      // A stack of all events to be dispatched, to ensure the priority order
+      var eventStack = this.eventStack;
+
+      // Order of events stack execution, leave event isn't here because it executes in the
+      // recursiveDispatcher, for one loop less
+      [CHANGE, ENTER, MATCH].forEach(function(type){
+        // Execute the leave stack of events
+        while (eventStack[type].length > 0) {
+          var events = eventStack[type][0].events;
+          var element = eventStack[type][0].element;
```
Comment (romuloab) [inline src/push-state-tree.js]: nab

``` js
let {events, element} = eventStack[type][0];
```

**Comment by romuloab** [comment]
It's really hard to follow the PR. The changes confused the diff algo, so it looks like completely random.

Also, one build failed. Is that expected?

File: src/push-state-tree.js (hunk 1 above)
Comment (gartz) [inline src/push-state-tree.js]: good point

File: src/push-state-tree.js (hunk 2 above)
Comment (gartz) [inline src/push-state-tree.js:717]: in this case I want methods from any prototype level, so I think it's ok, since they might change the API in the future we don't know if it will be in the instance level or prototype level...

File: src/push-state-tree.js (hunk 3 above)
Comment (gartz) [inline src/push-state-tree.js]: nice, good one! :+1:

**Comment by gartz** [comment]
Actually it's the coveralls that is failing sometimes, their service is returning http error to Travis CI... I need to see how to ignore such error, because it's making a mess in the github build result.
Thanks for the review, I know this one it's hard, and it's getting worse, version 0.15 will have tons of changes.
